### ISSUE TITLE
Compare dates in query

### ISSUE THREAD

**Issue by ddeboer** [netwerk-digitaal-erfgoed/register#45]
To find the the URLs that must be re-crawled (#3), I need to find URLs for which `schema:datedPosted` is > x hours (say 24 hours) ago.

This [should work](https://www.w3.org/TR/sparql11-query/#expressions), but doesn’t:

https://triplestore.netwerkdigitaalerfgoed.nl/sparql?name=Date%20comparison&infer=true&sameAs=true&query=PREFIX%20schema%3A%20%3Chttp%3A%2F%2Fschema.org%2F%3E%0A%0ASELECT%20*%20WHERE%20%7B%0A%09%3Fs%20%3Fp%20%3Fo%20.%20%3Fs%20schema%3AdatePosted%20%3Fdate.%0A%20%20%20%20FILTER%20(%3Fdate%20%3C%20%222030-02-03T13%3A52%3A58.417Z%22%5E%5E%3Cxsd%3AdateTime%3E)%0A%7D

**Comment by coret** [comment]
```
PREFIX schema: <http://schema.org/>
SELECT * WHERE {
	?s schema:datePosted ?date.
    FILTER (STR(?date) < "2021-02-03")
}
```
Works, but not nice.

It seems some ?date are xsd:DateTime and some are strings (leave the FILTER part to see what I mean).

**Comment by coret** [comment]
Can't seem to cast the datePosted to an xsd:dateTime:
```
PREFIX schema: <http://schema.org/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT * WHERE {
	?s schema:datePosted ?date
    BIND(xsd:dateTime("2021-02-01T20:37:37.469Z") as ?a)
    BIND("2021-02-01T20:37:37.469Z"^^xsd:dateTime as ?b)
    BIND(xsd:dateTime(?date) as ?c)
    BIND(coalesce(xsd:dateTime(?date), ?date) as ?d)
}
```

**Comment by coret** [comment]
I've made a query on the same ANL dataset in GraphDB and Virtuoso (TriplyDB) to filter results on date.
```
SELECT * WHERE {
  ?sub schema:dateModified ?date .
  FILTER (?date < "2020-10-26"^^schema:Date)
} LIMIT 10
```
The [query on GraphDB](https://triplestore.netwerkdigitaalerfgoed.nl/sparql?name=Overzicht%20van%20recent%20toegevoegde%20datasets&infer=true&sameAs=true&query=PREFIX%20schema%3A%20%3Chttp%3A%2F%2Fschema.org%2F%3E%0A%0ASELECT%20*%20FROM%20%3Chttp%3A%2F%2Fwww.archieven.nl%2Fdatasets-generated%3E%0AWHERE%20%7B%20%0A%09%3Fsub%20schema%3AdateModified%20%3Fdate%20%0A%20%20%20%20FILTER%20(%3Fdate%20%3C%20%222020-10-26%22%5E%5Eschema%3ADate)%0A%7D%20LIMIT%) has no results, the [TriplyDB version](https://data.netwerkdigitaalerfgoed.nl/coret/-/queries/DateFilterTestCase/1) (needs login) does have results, go figure?

**Comment by ddeboer** [comment]
I’ve implemented the string workaround in #49.

**Comment by coret** [comment]
String workaround suffices for our usecase.
As this does raise a question about GraphDB I will refer Ontotext support to this issue.
